### PR TITLE
Upgrade `@netlify/config`

### PR DESCRIPTION
**Pull request by ehmicky**
Due to https://github.com/netlify/build/pull/868, the latest major release of `@netlify/config` had to merge the method for resolving the configuration and the method for retrieving the configuration file path.

This breaking change requires the CLI to be updated accordingly.

### CODE REVIEW

File: src/commands/build/index.js [hunk 1]
```diff
@@ -24,26 +23,15 @@ class BuildCommand extends Command {
...
+    const {
+      site: { id: siteId }
+    } = this.netlify
     const {
       flags: { dry = false, context }
     } = this.parse(BuildCommand)
     const [token] = this.getConfigToken()
-
-    // Try current directory first, then site root
-    const config = (await getConfigPath()) || (await getConfigPath(undefined, this.netlify.site.root))
```
Comment (ehmicky) [inline src/commands/build/index.js]: This line is now done under the hood by `@netlify/config` (through `@netlify/build`) when the `config` option is `undefined`.

**Comment by RaeesBhatti** [comment]
[This function](https://github.com/DavidWells/configorama/blob/60eb164014329d1c4e884486865e1de8d1c1f74f/lib/main.js#L65) is causing memory leaks in Netlify CLI because we try to parse config every time the config file changes and it adds new event listeners. I would like to get more clarity on why we're adding event handlers. Or why we aren't using `configorama.sync` in `@netlify/config` (which avoids those event handlers).
cc @ehmicky @DavidWells

**Comment by ehmicky** [comment]
The latest version of `@neltify/config` (which is upgraded by this PR) does not use `configorama` anymore, so that should solve this problem.

**Comment by RaeesBhatti** [comment]
Cool, thanks @ehmicky

**Comment by ehmicky** [comment]
Please note `@netlify/config` also needs to be upgraded in `netlify-redirect-parser`: https://github.com/netlify/netlify-redirect-parser/blob/cc359c546926a456f8cc03957182ec7532ef2914/netlify-config-parser.js#L12 to prevent those leaks.

I can work on that.

**Comment by erquhart** [comment]
@ehmicky any update on this?

**Comment by ehmicky** [comment]
I was waiting for PR approvals to merge it.

File: src/commands/build/index.js [hunk 2]
```diff
@@ -24,26 +18,20 @@ class BuildCommand extends Command {
...
-      flags: { dry = false, context }
+      site: { id: siteId },
+      cachedConfig
+    } = this.netlify
+    // We have already resolved the configuration using `@netlify/config`
+    // This is stored as `this.netlify.cachedConfig` and can be passed to
+    // `@netlify/build --cachedConfig`.
+    const cachedConfigOption = JSON.stringify(cachedConfig)
+    const {
+      flags: { dry = false }
```
Comment (ehmicky) [inline src/commands/build/index.js]: `context` information is included in `cachedConfig`.

**Pull request merged**